src/language_features/cquery.rs

```diff
@@ -0,0 +1,208 @@
+use types::*;
+use context::*;
+use url::Url;
+use languageserver_types::{Range, NumberOrString};
+use url_serde;
+use serde;
+
+enum_from_primitive!{
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub enum StorageClass {
+    Invalid = 0,
+    No = 1,
+    Extern = 2,
+    Static = 3,
+    PrivateExtern = 4,
+    Auto = 5,
+    Register = 6
+}
+}
+
+impl<'de> serde::Deserialize<'de> for StorageClass {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use enum_primitive::FromPrimitive;
+
+        let i = try!(u8::deserialize(deserializer));
```

Comment:
Could be made even more ergonomic with `?` operator: `let i = u8::deserialize(deserializer)?;`

Reply:
Oh, i just copied this from the lsp-types package

Reply:
Probably it's because `?` operator was stabilized recently and more mature codebases have `try!`.